nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs

```diff
@@ -114,21 +118,37 @@ public static IEnumerable<string> GetProjectPathsFromProject(string projFilePath
                     continue;
                 }
 
-                projectPath = PathHelper.GetFullPathFromRelative(folderPath, projectPath);
+                Matcher matcher = new Matcher();
+                matcher.AddInclude(projectReference.Include);
+                
+                var fullPath = PathHelper.GetFullPathFromRelative(folderPath, projectPath);
+
+                string searchDirectory = folderPath;
 
-                var projectExtension = Path.GetExtension(projectPath).ToLowerInvariant();
-                if (projectExtension == ".proj")
+                IEnumerable<string> files = matcher.GetResultsInFullPath(searchDirectory);
```

Comment:
We need a test in UpdateWorker.DirsProj.cs for this behavior.

Reply:
Added a couple of tests.